### DISCUSSION TITLE
Issues on TRAMP connections

### DISCUSSION THREAD

**Discussion by alexluigit** [alexluigit/dirvish#124] [General]
If you met any issues when using dirvish over a TRAMP connection, please report it here.  You should at least post the TRAMP and dirvish config you are using.  Since I don't have access to the host you're using,  so you probably need to do some debugging by your self, it would be best if you can pin down the cause or send over a PR directly.  This discussion can be seen as the continuation of https://github.com/alexluigit/dirvish/issues/37

**Comment by alexluigit** [comment]
> > @tdstoff , by 'hangs indefinitely' did you mean it hangs after invoking `dirvish-yank`?
> 
> No, actually. I set the following:
> 
> ```
>   (add-to-list 'tramp-connection-properties
>                (list (regexp-quote "/ssh:myuser@host:")
>                      "direct-async-process" t))
> ```
> 
> and notice that file sizes are then shown like in this screenshot: <img alt="image" width="589" src="https://user-images.githubusercontent.com/43076118/182354680-ab32f7fb-b4cd-407c-a36b-36d6b6bd8d07.png">
> 
> But within ~10 seconds or so of using the buffer, I'm stuck in a hang that incessant `C-g`'s can't remedy. No use of `dirvish-yank` required to get this behavior.
> 
> Edit: I reproduced this three times in the same remote directory, then tried accessing the remote host at a different location and it seems to be working at the moment, even `dirvish-yank`. I can keep fuddling with it and see if there's some consistent cause. If you have any suggestions off the top of your head, they're appreciated.

> Does the above setting `"direct-async-process" t` for a remote connection break shell/terminal buffers in Emacs for anyone else? After I do this, I get the message `stty: standard input: Inappropriate ioctl for device` in my shell buffer, and vterm processes immediately die upon spawning. 

Yes, I observed the same behavior. And if I remembered correctly, it used to work fine in the past.  It looks like a bug in Emacs 29?

**Reply by tdstoff** [reply]
Yes, I recently upgraded to Emacs 29 as well. Any ideas on how to dig into this?

**Reply by alexluigit** [reply]
I'm not a TRAMP expert and have absolutely no clue on this.  Perhaps you should M-x report-emacs-bug

**Reply by ParetoOptimalDev** [reply]
@tdstoff Always check `info` from within emacs first... though I'll admit I had to force myself at first since it didn't *feel* like it would be as fruitful as a "quick web search".

Here's how I checked info in emacs to answer your question:

`c-h R tramp<RET>`

FYI without keyboard shortcuts that is `M-x info-display-manual <RET> tramp <RET>`

Then while tempted to search directly, I find it useful to browse the table of contents and see `* Traces and Profiles::         How to Customize Traces.`.

Navigating to it I see:

```
TRAMP messages are raised with verbosity levels ranging from 0 to 10.
TRAMP does not display all messages; only those with a verbosity level
less than or equal to ‘tramp-verbose’.

   The verbosity levels are

    0 silent (no TRAMP messages at all)
 1 errors
 2 warnings
 3 connection to remote hosts (default verbosity)
 4 activities
 5 internal
 6 sent and received strings
 7 connection properties
 8 file caching
 9 test commands
10 traces (huge)
11 call traces (maintainer only)
```

Then if you set `tramp-verbose` to 6 or 7, you should see enough information to debug your issue.

Hope this helps!
